app/models/setting.rb

```diff
@@ -242,9 +243,16 @@ def self.create_existing(s, opts)
       attrs = column_check([:default, :description, :full_name, :encrypted])
       to_update = Hash[opts.select { |k, v| attrs.include? k }]
       to_update[:value] = readonly_value(s.name.to_sym) if s.has_readonly_value?
-      s.update(to_update)
+      # default is converted to yaml so we need to convert the yaml here too,
+      # in order to check, if the update of default is needed
+      # if it is the same, we don't try to update default, it would trigger
+      # update query for every setting
+      to_update.delete(:default) if to_update[:default].to_yaml.strip == s[:default]
+      s.attributes = to_update
+      s.save if s.changed?
```

Comment:
also instead of calling `s.update` blindly, if we assign the attributes and check if update is needed, we avoid `Exist` SQL queries for each and every setting